src/hotspot/share/oops/fieldInfo.hpp

```diff
@@ -269,16 +266,34 @@ class FieldInfoStream : AllStatic {
   friend class JavaFieldStream;
   friend class FieldStreamBase;
   friend class ClassFileParser;
+  friend class FieldInfoReader;
+
+ private:
+  // Don't generate the table for small classes at all.
+  static const int SEARCH_TABLE_THRESHOLD = 16;
```

Comment:
One way to create a stress test is to move this to globals.hpp, as a debug-only variable, something like `BinarySearchThreshold` (default 16).  It's OK not to name it more specifically, if (as I believe) this algorithm is destined to be pulled out and used in several places.  Individual uses can scale the number if need be.